src/valid/expression.rs

```diff
@@ -560,16 +561,9 @@ impl super::Validator {
                                 ))
                             }
                         };
-                        let needs_index = match class {
-                            crate::ImageClass::Storage { .. } => false,
-                            _ => true,
```

Comment:
we appear to be losing this check, aren't we?
i.e. a situation where either the sample index or the level is required but not provided.
The old code would detect this, but the new code doesn't care.